src/api/c++/z3++.h

```diff
@@ -593,23 +675,18 @@ namespace z3 {
     /**
        \brief A Z3 sort (aka type). Every expression (i.e., formula or term) in Z3 has a sort.
     */
-    class sort : public ast {
+    class sort final : public ast {
     public:
         sort(context & c):ast(c) {}
         sort(context & c, Z3_sort s):ast(c, reinterpret_cast<Z3_ast>(s)) {}
         sort(context & c, Z3_ast a):ast(c, a) {}
-        sort(sort const & s):ast(s) {}
         operator Z3_sort() const { return reinterpret_cast<Z3_sort>(m_ast); }
 
         /**
            \brief retrieve unique identifier for func_decl.
          */
         unsigned id() const { unsigned r = Z3_get_sort_id(ctx(), *this); check_error(); return r; }
 
-        /**
-           \brief Assign sort s to this
-        */
-        sort & operator=(sort const & s) { return static_cast<sort&>(ast::operator=(s)); }
```

Comment:
Returning a `static_cast` of the parent's operator is unnecessary compared to `return *this`. Any compiler should be able to able to optimize away the `*this` operation.

More important, this may introduce undefined behavior. Specifically, there is no guarantee that the parent `=` operator will return `this`; it is done by convention but not required. In case the operator returns something else, we are now statically downcasting an unknown pointer to a type it may not be, and one which we may not own.